Model/Quote/Address/Total/Surcharge.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * @by SwiftOtter, Inc. 3/30/17
+ * @website https://swiftotter.com
+ **/
+
+namespace SwiftOtter\ShippingSurcharge\Model\Quote\Address\Total;
+
+use \SwiftOtter\ShippingSurcharge\Model\Surcharge as SurchargeModel;
+
+class Surcharge extends \Magento\Quote\Model\Quote\Address\Total\AbstractTotal
+{
+    /**
+     * @var \Magento\Framework\Pricing\PriceCurrencyInterface
+     */
+    private $priceCurrency;
+
+    /**
+     * @var \SwiftOtter\ShippingSurcharge\Api\Calculator\SurchargeCalculatorInterface
+     */
+    private $surchargeCalculator;
+
+    /**
+     * @var \Magento\Quote\Api\Data\TotalSegmentExtensionFactory
+     */
+    private $cartExtensionFactory;
+
+
+    public function __construct(
+        \Magento\Framework\Pricing\PriceCurrencyInterface $priceCurrency,
+        \SwiftOtter\ShippingSurcharge\Api\Calculator\SurchargeCalculatorInterface $surchargeCalculator,
+        \Magento\Quote\Api\Data\CartExtensionFactory $cartExtensionFactory
+    )
+    {
+        $this->setCode(SurchargeModel::SURCHARGE);
+        $this->priceCurrency = $priceCurrency;
+        $this->surchargeCalculator = $surchargeCalculator;
+        $this->cartExtensionFactory = $cartExtensionFactory;
+    }
+
+    public function collect(
+        \Magento\Quote\Model\Quote $quote,
+        \Magento\Quote\Api\Data\ShippingAssignmentInterface $shippingAssignment,
+        \Magento\Quote\Model\Quote\Address\Total $total
+    ) {
+        parent::collect($quote, $shippingAssignment, $total);
+
+        $surchargeAmount = $this->calculateSurcharge($quote);
+
+        if ($surchargeAmount) {
```

Comment:
You can save three lines by moving the definition into the `if` statement, like `if ($surchargeAmount = $this->calculateSurcharge($quote))`. It will do more than one thing, but I believe that it's pretty easily understandable and very clean and, thus, worth the trade off.